docs/guides/location-service/fragments/setting-up-your-app-android.md

```diff
@@ -0,0 +1,259 @@
+
+This section provides you an example of using the Amazon Location Service APIs. In this example, we will first create a Place Index in the Amazon Location Service console and use the APIs to search for places.  
+
+### Creating a new place index
+
+1. Open the [Amazon Location Service console](https://console.aws.amazon.com/location/places/home#/create) to create a place index.
+1. Enter **MyPlaceIndex** in **Name**.
+1. Press **Create place index**
+
+    ![Amazon Location Service - Create place index](~/images/als/create-place-index.png)
+
+1. Note the Amazon Resource Name (ARN) of your place index. This will start with *arn:aws:geo* as in the below screenshot.
+
+    ![Amazon Location Service - Place index](~/images/als/my-place-index.png)
+
+### Allow Guest users access to the place index
+
+Now create an inline policy that will give guest users of your application access to the Place Index resource you just created in Amazon Location Service.
+
+1. Navigate to the root of your project and run the following command:
+
+    ```bash
+    amplify console auth
+    ```
+
+1. Select **Identity Pool** for **Which console?** when prompted.
+1. You will be directed to the Amazon Cognito console. Click on **Edit identity pool** in the top right corner of the page.
+1. Open the drop down for **Unauthenticated identities**, choose **Enable access to unauthenticated identities**, and then press **Save Changes**.
+1. Click on **Edit identity pool** once more. Make a note of the name of the Unauthenticated role. For example, `amplify-<project_name>-<env_name>-<id>-unauthRole`.
+1. Open the [AWS Identity and Access Management (IAM) console](https://console.aws.amazon.com/iam/home#/roles) to manage Roles.
+1. In the **Search** field, enter the name of your Unauthenticated role noted above and click on it.
+1. Click **+Add inline policy**, then click on the **JSON** tab.
+1. Fill in the `[ARN]` placeholder with the ARN of your place index which you noted above and replace the contents of the policy with the below.
+
+    ```json
+{
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": "geo:SearchPlaceIndexForText",
+                "Resource": "[ARN]"
+            }
+        ]
+}
+    ```
+
+1. Click on the **Review policy** button.
+1. In the **Name** field, enter *LocationTracker*.
+1. Click on the **Create policy** button. You have now successfully added authentication to your Android app.
+
+### Searching for places
+
+Here is how you can search for places using the place index you just created:
+
+<amplify-block-switcher>
+
+<amplify-block name="Java">
+
+```java
+AWSMobileClient.getInstance().initialize(getApplicationContext(), new Callback<UserStateDetails>() {
+    @Override
+    public void onResult(UserStateDetails userStateDetails) {
+        ExecutorService executorService = Executors.newFixedThreadPool(4);
```

Comment:
question, why is this needed and why 4?

Reply:
Previously, we had a construct like:
```java
new Thread(() -> {
    // do location stuff ...
}).start();
```
Android's official documentation [suggested doing background tasks on an `Executor`](https://developer.android.com/guide/background/threading#executing-in-a-background-thread), and chooses a `Executors.newFixedThreadPool(4)` somewhat at random.

I think the goal here was to show some background work in a way that was close to the platforms docs.